src/blob_file_builder.cc

```diff
@@ -33,19 +33,41 @@ BlobFileBuilder::BlobFileBuilder(const TitanDBOptions& db_options,
     return;
 #endif
   }
+  block_size_ = cf_options.enable_punch_hole_gc ? cf_options.block_size : 0;
   WriteHeader();
 }
 
+void BlobFileBuilder::FillBlockWithPad() {
+  if (block_size_ == 0 || file_->GetFileSize() % block_size_ == 0) {
+    return;
+  }
+  static const std::string pad = std::string(4096, 0);
+
+  size_t pad_size = block_size_ - (file_->GetFileSize() % block_size_);
+  while (pad_size > pad.size()) {
+    status_ = file_->Append(pad);
+    if (!ok()) {
+      return;
+    }
+    pad_size -= pad.size();
+  }
+  status_ = file_->Append(Slice(pad.data(), pad_size));
+}
+
 void BlobFileBuilder::WriteHeader() {
   BlobFileHeader header;
   header.version = blob_file_version_;
   if (cf_options_.blob_file_compression_options.max_dict_bytes > 0) {
-    assert(blob_file_version_ == BlobFileHeader::kVersion2);
+    assert(blob_file_version_ >= BlobFileHeader::kVersion2);
     header.flags |= BlobFileHeader::kHasUncompressionDictionary;
   }
+  header.block_size = block_size_;
   std::string buffer;
   header.EncodeTo(&buffer);
   status_ = file_->Append(buffer);
+  if (block_size_ > 0) {
+    FillBlockWithPad();
+  }
```

Comment:
nit: `FillBlockWithPad` already checks `block_size_` internally. Another way to write it is:
```suggestion
  MaybeFillBlockWithPad();
```